### PR TITLE
Add icon support for files without extensions

### PR DESCRIPTION
**Pull request by Imgkl**
Release Notes:

- Added support for showing file icons for files without suffixes.

Before:

image

After:

Screenshot 2024-02-27 at 1 51 20 AM

This screenshot is to show if the file has extension, then the extension takes precedence.

image

<br>

- Added icons for
    - Docker - https://www.svgrepo.com/svg/473589/docker
    - License - https://www.svgrepo.com/svg/477704/license-1
    - Heroku - https://www.svgrepo.com/svg/341904/heroku
 
 - Updated tests

### CODE REVIEW

File: assets/icons/file_icons/docker.svg
Comment (maxdeviant) [inline assets/icons/file_icons/docker.svg:1]: Can we change the colors on this icon to black like our other ones?

**Comment by Imgkl** [comment]
@maxdeviant done. 😄

File: assets/icons/file_icons/docker.svg
Comment (Imgkl) [reply assets/icons/file_icons/docker.svg:1]: Sure

File: assets/icons/file_icons/license.svg
Comment (maxdeviant) [inline assets/icons/file_icons/license.svg:1]: I'm not a huge fan of this icon for license files.

Do we even need a different icon for license files?

Comment (Stzx) [reply assets/icons/file_icons/license.svg:1]: Maybe https://www.svgrepo.com/svg/371962/license would be more appropriate?

Comment (Imgkl) [reply assets/icons/file_icons/license.svg:1]: @Stzx yeah this makes more sense to me. 

@maxdeviant the reason for license to have it's own icon is, I'm coming from VSC, there we have it, thanks to [Material Icon Theme](https://github.com/PKief/vscode-material-icon-theme/blob/main/icons/certificate.svg). Plus I feel it provides better context to the files when we glance the file pane. Your thoughts on icon suggested by @Stzx?

Comment (maxdeviant) [reply assets/icons/file_icons/license.svg:1]: In the default VS Code icon theme there are no icons for license files:

Screenshot 2024-02-27 at 10 29 59 AM

Neither of the proposed icons really feel like a good fit for Zed, so I think we should forego icons for license files entirely.

Comment (Imgkl) [reply assets/icons/file_icons/license.svg:1]: Sure. I'll change this in a bit.

Comment (Imgkl) [reply assets/icons/file_icons/license.svg:1]: image

Updated the code.

File: crates/util/src/paths.rs [hunk 1]
```diff
@@ -100,17 +100,21 @@ impl<T: AsRef<Path>> PathExt for T {
...
             return file_name.strip_prefix('.');
         }
 
-        self.as_ref()
-            .extension()
-            .and_then(|extension| extension.to_str())
+        let extension = path.extension().and_then(|e| e.to_str());
+        let file_stem = path.file_stem()?.to_str()?;
+
+        match extension {
```
[the remark this answers is not in the document]
Comment (Imgkl) [reply crates/util/src/paths.rs]: Added lazy evaluation for stem string.

**Comment by maxdeviant** [comment]
@Imgkl Thanks for working through this!

**Pull request merged**